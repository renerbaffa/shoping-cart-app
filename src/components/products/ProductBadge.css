.ProductBadge-container {
  position: absolute;
  top: 5px;
  right: 12px;
  background-color: #FF6D3E;
  padding: 0px 7px;
  border-radius: 41%;
  color: white;
}