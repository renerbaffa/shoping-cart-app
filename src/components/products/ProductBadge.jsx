import React from 'react';
import PropTypes from 'prop-types';

import './ProductBadge.css';

export const ProductBadge = ({ quantity }) => (
  <div className="ProductBadge-container">
    {quantity > 999 ? '99+' : quantity}
  </div>
);

ProductBadge.propTypes = {
  quantity: PropTypes.number,
};

ProductBadge.defaultProps = {
  quantity: 0,
};

export default ProductBadge;
