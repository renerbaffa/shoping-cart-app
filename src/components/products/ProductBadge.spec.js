import React from 'react';
import renderer from 'react-test-renderer';

import { ProductBadge } from './ProductBadge';

describe('<ProductBadge />', () => {
  it('should render correctly', () => {
    const productBadge =
      renderer.create(<ProductBadge quantity={99} />).toJSON();
    expect(productBadge).toMatchSnapshot();
  });

  it('should render 99+ when quantity is grater than 999', () => {
    const productBadge =
      renderer.create(<ProductBadge quantity={9999} />).toJSON();
    expect(productBadge).toMatchSnapshot();
  });
});
