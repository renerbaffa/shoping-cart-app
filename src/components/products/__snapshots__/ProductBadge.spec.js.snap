// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductBadge /> should render 99+ when quantity is grater than 999 1`] = `
<div
  className="ProductBadge-container"
>
  99+
</div>
`;

exports[`<ProductBadge /> should render correctly 1`] = `
<div
  className="ProductBadge-container"
>
  99
</div>
`;
